cmd/goal/application.go

```diff
@@ -995,3 +1023,100 @@ var infoAppCmd = &cobra.Command{
 		}
 	},
 }
+
+var methodAppCmd = &cobra.Command{
+	Use:     "method",
+	Short:   "Invoke a method",
+	Long:    `Invoke a method in an App (stateful contract) with an application call transaction`,
+	Args:    validateNoPosArgsFn,
+	PreRunE: validateNoPosArgsFn,
+	Run: func(cmd *cobra.Command, args []string) {
+		var applicationArgs [][]byte
+
+		// insert the method selector hash
+		hash := sha512.Sum512_256([]byte(method))
+		applicationArgs = append(applicationArgs, hash[0:4])
+
+		// parse down the ABI type from method signature
+		argTupleTypeStr, retTypeStr, err := abi.ParseMethodSignature(method)
+		if err != nil {
+			reportErrorf("cannot parse method signature: %v", err)
+		}
+		err = abi.ParseArgJSONtoByteSlice(argTupleTypeStr, methodArgs, &applicationArgs)
+		if err != nil {
+			reportErrorf("cannot parse arguments to ABI encoding: %v", err)
+		}
+
+		// copy-paste code from callAppCmd
+		dataDir := ensureSingleDataDir()
+		client := ensureFullClient(dataDir)
+
+		// Parse transaction parameters
+		_, appAccounts, foreignApps, foreignAssets := getAppInputs()
+
+		tx, err := client.MakeUnsignedAppNoOpTx(appIdx, applicationArgs, appAccounts, foreignApps, foreignAssets)
```

Comment:
Instead of always assuming OnCompletion is no-op, it would be nice if you could use `createOnCompletion` to set the OnCompletion value (and probably rename that variable to be more general).

Reply:
I am not quite sure if the current code change is the correct practice, but I tried to retrieve the `onCompletion` from commandline, and instantiate a transaction (unsigned) from `MakeUnsignedApplicationCallTx`.
It seems some `onCompletion` option is not applicable here?